internal/framework/with_import_by_arn.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package framework
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// WithImportByARN is intended to be embedded in resources which import state via the "id" attribute.
+// See https://developer.hashicorp.com/terraform/plugin/framework/resources/import.
+type WithImportByARN struct{}
+
+func (w *WithImportByARN) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) { // nosemgrep:ci.semgrep.framework.with-import-by-id
```

Comment:
You can remove the `nosemgrep`, since it's only triggered if the ID attribute is `names.AttrID`

```suggestion
func (w *WithImportByARN) ImportState(ctx context.Context, request resource.ImportStateRequest, response *resource.ImportStateResponse) {
```